compiler/dyno/lib/resolution/Resolver.cpp

```diff
@@ -1239,6 +1349,77 @@ void Resolver::exit(const Dot* dot) {
   // TODO: resolve field accessors / parenless methods
 }
 
+bool Resolver::enter(const uast::New* nw) {
+  return true;
+}
+
+void Resolver::resolveNewForClass(const uast::New* node,
+                                  const types::ClassType* classType) {
+  assert(false && "Not handled yet!");
+}
+
+void Resolver::resolveNewForRecord(const uast::New* node,
+                                   const RecordType* recordType) {
+  ResolvedExpression& re = byPostorder.byAst(node);
+
+  if (node->management() != New::DEFAULT_MANAGEMENT) {
+    auto managementStr = New::managementToString(node->management());
+    auto recordNameStr = recordType->name().c_str();
+    context->error(node, "Cannot use new %s with record %s",
+                         managementStr,
+                         recordNameStr);
+  } else {
+
+    auto qt = QualifiedType(QualifiedType::VAR, recordType);
+    re.setType(qt);
+  }
+}
+
+void Resolver::exit(const uast::New* node) {
+
+  // Fetch the pieces of the type expression.
+  const Expression* typeExpr = node->typeExpression();
+  ResolvedExpression& reTypeExpr = byPostorder.byAst(typeExpr);
+  auto& qtTypeExpr = reTypeExpr.type();
+
+  // TODO: What about if the thing doesn't make sense/is 'UNKNOWN'?
+  if (qtTypeExpr.kind() != QualifiedType::TYPE) {
+    context->error(node, "'new' must be followed by a type expression");
+  }
+
+  // TODO: Set ErroneousType here.
+  if (qtTypeExpr.type()->isUnknownType()) {
+    assert(false && "Not handled yet!");
+  }
```

Comment:
Could we add a `isErroneous` and `isUnknown` to `QualifiedType` if they aren't there already (I think `isUnknown` might be)? Just to make this easy to check?

Reply:
We could add `QualifiedType::` `isErroneousType` and `isUnknownType` if they don't exist yet

Reply:
I added `isErroneousType` but did not add `isUnknownType` because we already have `isUnknown`.

Reply:
You have to be careful with `isUnknown` though because it will also return true for something with unknown intent/qualifier. Which may or may not be what you want.